src/io/hdf/hdf_archive.cpp

```diff
@@ -189,11 +192,12 @@ bool hdf_archive::is_group(const std::string& aname)
   }
 }
 
-hid_t hdf_archive::push(const std::string& gname, bool createit)
+void hdf_archive::push(const std::string& gname, bool createit)
 {
   hid_t g = is_closed;
   if (Mode[NOIO] || file_id == is_closed)
-    return is_closed;
+    throw std::runtime_error("Failed to open group \"" + gname +
+                             "\" because file is not open.  Expected file: " + possible_filename_);
```

Comment:
probably there is a behavior change with `Mode[NOIO] ` which is intended to treat every operation as no-op.
So probably
```
if (Mode[NOIO]) return;
if (file_id == is_closed))
  throw
```